### PR TITLE
Fix Transactionbuilder and keypair

### PR DESCRIPTION
**Pull request by marktoda**
The transactionbuilder and keypair for ETH were broken with the
following bugs:
- keypair would not generate addresses from an extended key, simply due
to not handling the case
- transactionbuilder would throw if not given a counter, even though
there is a default

This commit fixes the above issues and adds some tests

Ticket: BG-20990

### CODE REVIEW

File: src/coin/eth/transactionBuilder.ts [hunk 1]
```diff
@@ -143,20 +141,36 @@ export class TransactionBuilder extends BaseTransactionBuilder {
...
+        }
+
+        if (this._walletOwnerAddresses.length !== 3) {
+          throw new BuildTransactionError(
+            `Invalid transaction: wrong number of owners -- required: 3, ` +
+              `found: ${this._walletOwnerAddresses.length}`,
+          );
+        }
+
+        if (this._counter === undefined) {
```
Comment (argjv) [inline src/coin/eth/transactionBuilder.ts:165]: Is this the nonce? Counter is the Tezos name for nonces.
Also, if it is a string, isn't it better to check for `if (!this._counter)`?

File: src/coin/eth/transactionBuilder.ts [hunk 2]
```diff
@@ -143,20 +141,36 @@ export class TransactionBuilder extends BaseTransactionBuilder {
...
+          throw new BuildTransactionError('Invalid transaction: missing address counter');
+        }
+
+        if (this._sourceAddress === undefined) {
```
Comment (argjv) [inline src/coin/eth/transactionBuilder.ts:169]: Ditto.

File: src/coin/eth/utils.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Buffer } from 'buffer';
+import { HDNode } from 'bitgo-utxo-lib';
```
Comment (argjv) [inline src/coin/eth/utils.ts:1]: Why is this change needed?

File: src/coin/eth/transaction.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@
  * Ethereum transaction model
  */
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import transaction from 'ethereumjs-tx/dist/transaction';
```
Comment (argjv) [inline src/coin/eth/transaction.ts:5]: Is this needed? It is not used in this PR

File: src/coin/eth/transactionBuilder.ts (hunk 1 above)
Comment (marktoda) [reply src/coin/eth/transactionBuilder.ts:165]: no, it's a number and it can be 0. that is one of the bugs I'm fixing here

File: src/coin/eth/utils.ts (hunk 3 above)
Comment (marktoda) [reply src/coin/eth/utils.ts:1]: we use xpubs for ETH, previously it would throw if given an xpub

File: src/coin/eth/transaction.ts (hunk 4 above)
Comment (marktoda) [reply src/coin/eth/transaction.ts]: removed

File: src/coin/eth/utils.ts (hunk 3 above)
Comment (marktoda) [reply src/coin/eth/utils.ts]: removed

File: src/coin/eth/transactionBuilder.ts (hunk 1 above)
Comment (marktoda) [reply src/coin/eth/transactionBuilder.ts:165]: I'd prefer the name is nonce myself, but I don't want to step over simtlix team's toes as they probably have other PRs built on that name, and it's fine for consistency with xtz. I just wanted to fix the bug to unblock myself

Comment (marktoda) [reply src/coin/eth/transactionBuilder.ts:165]: for context, I tried to send a tx with nonce 0 (First tx on the wallet) and it failed because of the `(!this._counter)` check.

File: src/coin/eth/transactionBuilder.ts (hunk 2 above)
Comment (marktoda) [reply src/coin/eth/transactionBuilder.ts]: fixed!
